OpenRA.Game/Server/Connection.cs

```diff
@@ -120,6 +126,59 @@ public void ReadData(Server server)
 					}
 				}
 		}
+
+		public void SendData(byte[] data, Server server)
+		{
+			BufferData(data);
+			Flush(server);
+		}
+
+		public void BufferData(byte[] data)
+		{
+			SendBuffer.AddRange(data);
+		}
+
+		public void Flush(Server server)
+		{
+			if (SendBuffer.Count == 0)
+				return;
+
+			var data = SendBuffer.ToArray();
+			SendBuffer.Clear();
+
+			try
+			{
+				InnerSend(data);
+			}
+			catch (Exception e)
+			{
+				server.DropClient(this);
+				Log.Write("server", "Dropping client {0} because dispatching orders failed: {1}", PlayerIndex, e);
+			}
+		}
+
+		void InnerSend(byte[] data)
```

Comment:
Would it make sense for us to try and finish and merge my prototype server freeze fix before we proceed here?

Reply:
#19429